src/components/SortableArea.tsx

```diff
@@ -0,0 +1,105 @@
+import {
+  closestCenter,
+  DndContext,
+  DragEndEvent,
+  KeyboardSensor,
+  PointerSensor,
+  UniqueIdentifier,
+  useSensor,
+  useSensors,
+} from "@dnd-kit/core";
+import {
+  arrayMove,
+  SortableContext,
+  sortableKeyboardCoordinates,
+  verticalListSortingStrategy,
+} from "@dnd-kit/sortable";
+import { CSS } from "@dnd-kit/utilities";
+import { useSortable } from "@dnd-kit/sortable";
+import { Drag } from "./icons";
+
+type Props<T> = {
+  children: (item: T, index: number) => React.ReactNode;
+  setItems: (items: T[]) => void;
+  items: T[];
+  disableDragByIndex?: number;
```

Comment:
it's unclear what is the purpose of this.

I mean reading the code, I can understand that it disabled the drag of a single item.

I am guessing that this is meant to disable the dragging of the "catch-all" item? :thinking: 

but it's kinda an odd way to do this :thinking: 

1. Does the catch all item (if it's not draggable) even have to be part of the draggable area? I mean is it not always the last item?
2. If does **has to** be in the draggable area, I think it would make sense to extend the generic type  for items (`T`) to have a "isDraggable" attribute instead

Reply:
we decided avoid to drag yesterday, and I didn't have time to refactor this...moving out from draggable area means a big refactor that I would avoid now. 
I agree on adding isDraggable prop into the list, but I would do in the next PR, cause I would like to merge it...it is a big PR and I would avoid conflicts with main